### PR TITLE
feat(rating): move rating component from angularComp to ng-comp

### PR DESCRIPTION
**Pull request by DanSallau**
Rating is a straight forward component and mostly inconsequential, simply moved from angular components to here. 

Preview https://github.sebank.se/pages/DesignLibrary/AngularComponents/#/rating

### CODE REVIEW

File: lib/src/rating/rating.component.html [hunk 1]
```diff
@@ -0,0 +1,48 @@
+<div class="rating-container" [ngClass]="className">
+    <div class="star-rating" [class.readonly]="readOnly" role="radiogroup" aria-labelledby="radioGroup">
```
Comment (yousifalraheem) [inline lib/src/rating/rating.component.html:2]: can you rename `readonly` class to `read-only`

Comment (yousifalraheem) [inline lib/src/rating/rating.component.html:2]: the `aria-labelledby` should be passed by the user if necessary, not hardcoded

File: lib/src/rating/rating.component.ts [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+    tabCounter: number = 0;
+
+    @ViewChildren("radioItemRefs") radios: QueryList<ElementRef>;
+
+    ngOnInit() {
+        if (this.max) {
+            this.randomIds = Array.apply(null, Array(this.max)).map((item: any) => {
+                const identifier = +Math.floor(Math.random() * 100) + new Date().getTime();
+                return "star-" + identifier;
+            });
```
Comment (yousifalraheem) [inline lib/src/rating/rating.component.ts:53]: can you use `randomId` from [frontend-tools](https://github.com/sebgroup/frontend-tools/wiki/randomId)

File: lib/src/rating/rating.component.ts [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+    /**
+     * Provides an array for ngFor to loop through
+     * @returns {Array<number>} An array of ones just to loop through it
+     */
+    getList(): Array<number> {
+        const list: Array<number> = new Array(this.max);
+        for (let i: number = 0; i < list.length; i++) {
+            list[i] = 1;
+        }
+        return list;
```
Comment (yousifalraheem) [inline lib/src/rating/rating.component.ts:66]:
```suggestion
        return Array(this.max).fill(1);
```

File: lib/src/rating/rating.component.ts [hunk 4]
```diff
@@ -0,0 +1,231 @@
...
+    }
+
+    setTabIndex(i: number): string {
+        if (i === 0) {
+            return "0";
+        }
+        return Math.floor(this.value) === i + 1 ? "0" : "-1";
```
Comment (yousifalraheem) [inline lib/src/rating/rating.component.ts:73]:
```suggestion
        return i === 0 ? "0" : Math.floor(this.value) === i + 1 ? "0" : "-1";
```

File: lib/src/rating/svgStarHollow.svg.html [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+    </defs>
+    <g>
+        <rect fill="none" id="canvas_background" height="6.61537" width="6.61537" y="-1" x="-1" />
+    </g>
+    <g>
+        <title *ngIf="title">{{title}}</title>
+        <g fill-rule="evenodd" fill="none" id="Page-1">
+            <g id="Icons-main-navigation/Star-unselected/Darkblue">
+                <g id="Icon">
+                    <mask transform="translate(2,2) " id="mask-unselected">
```
Comment (yousifalraheem) [inline lib/src/rating/svgStarHollow.svg.html:23]: I think we should remove all IDs since we are rendering at least 5 instances of either of the SVGs.

**Comment by mario-subo** [comment]
Have you made any changes from Angular Components that need my attention? Did you copy over the unit tests and are they passing?

File: lib/src/rating/rating.component.html (hunk 1 above)
Comment (DanSallau) [reply lib/src/rating/rating.component.html:2]: Moved it to the appropriate span element. It doesn't belong here.

File: lib/src/rating/rating.component.ts (hunk 3 above)
Comment (DanSallau) [reply lib/src/rating/rating.component.ts:66]: Not necessary.

Comment (yousifalraheem) [reply lib/src/rating/rating.component.ts:66]: It's OK, we have polyfills for that, I just don't think the function is necessary. If you don't like it you can even use this:
```javascript
"1".repeat(this.max).split("").map(i => +i)
```

**Pull request merged**